plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -2906,7 +2905,10 @@ private void tearDownVm(VirtualMachineMO vmMo) throws Exception {
 
     int getReservedMemoryMb(VirtualMachineTO vmSpec) {
         if (vmSpec.getDetails().get(VMwareGuru.VmwareReserveMemory.key()).equalsIgnoreCase("true")) {
-            return (int) (vmSpec.getMinRam() / ResourceType.bytesToMiB);
+            if(vmSpec.getDetails().get(VmDetailConstants.RAM_RESERVATION) != null){
+                float reservedMemory = (vmSpec.getMaxRam() * Float.parseFloat(vmSpec.getDetails().get(VmDetailConstants.RAM_RESERVATION)));
+                return (int) (reservedMemory / ResourceType.bytesToMiB);
+            }
```

Comment:
```suggestion
            }
            return (int) (vmSpec.getMinRam() / ResourceType.bytesToMiB);
```

By default reserve all memory